src/util.ts

```
@@ -29,13 +29,67 @@ function stringFromMap(map: Map<Stringable, Stringable>): string {
   return JSON.stringify(Object.fromEntries(Array.from(map.entries()).sort()));
 }
 
-export function getIgnoredUsers(): Set<string> {
-  const s = new Set<string>();
+export function shouldProcessPullRequestForUser(user: string): boolean {
   const IGNORED_USERS = core.getInput("ignored-users");
-  if (!IGNORED_USERS) {
-    return s;
+  const ONLY_USERS = core.getInput("only-users");
+  let ignoredUsers;
+  let onlyUsers;
+
+  if (IGNORED_USERS) {
+    ignoredUsers = getUserListAsSet(IGNORED_USERS);
   }
-  for (const username of IGNORED_USERS.split(",")) {
+  if (ONLY_USERS) {
+    onlyUsers = getUserListAsSet(ONLY_USERS);
+  }
+
+  if (!ignoredUsers && !onlyUsers) {
+    core.debug(
+      "No users defined in only-users or ignored-users. Proceeding with Clubhouse workflow..."
+    );
+    return true;
+  }
+
+  if (onlyUsers && ignoredUsers) {
+    core.setFailed(
+      "You have defined both ignored-users and only-users lists, please use one or the other. Cancelling Clubhouse workflow..."
+    );
+    return false;
+  }
```

This is only a problem if the same user is in `only-users` and `ignored-users`. I don't see anything wrong with having both inputs specified, as long as they contain different contents.

Also, this should raise an exception, not `return false`.

---

@singingwolfboy, I can account for both lists being populated, but was thinking that is someone is using the `only-users` input, they are explicitly telling the action they only want it to run for the specific users and to ignore all other users. Can you think of a use case where both would be populated?

---

It's a bit of an edge case, but I could see someone wanting to try out the new `only-users` input without deleting their existing `ignore-users` input. It's a bit strange to specify both, but as long as they don't actually conflict with each other, I don't see a reason to disallow it.

---

Now throwing an error for the the use case where the author is in both `only-users` and `ignored-users`. Otherwise, I am showing a debug message and proceeding through the logic as requested.